tket/src/MeasurementSetup/MeasurementSetup.cpp

```diff
@@ -107,4 +107,50 @@ std::string MeasurementSetup::to_str() const {
   return ss.str();
 }
 
+void to_json(
+    nlohmann::json &j, const MeasurementSetup::MeasurementBitMap &result) {
+  j["circ_index"] = result.get_circ_index();
+  j["bits"] = result.get_bits();
+  j["invert"] = result.get_invert();
+}
+
+void from_json(
+    const nlohmann::json &j, MeasurementSetup::MeasurementBitMap &result) {
+  result = MeasurementSetup::MeasurementBitMap(
+      j.at("circ_index").get<unsigned>(),
+      j.at("bits").get<std::vector<unsigned>>(), j.at("invert").get<bool>());
+}
+
+void to_json(nlohmann::json &j, const MeasurementSetup &setup) {
+  std::vector<std::pair<
+      QubitPauliString, std::vector<MeasurementSetup::MeasurementBitMap>>>
+      map_list;
+  for (const std::pair<
+           const QubitPauliString,
+           std::vector<MeasurementSetup::MeasurementBitMap>> &tensor_map :
+       setup.get_result_map()) {
+    map_list.push_back(tensor_map);
+  }
+  // sort the list for consistent serialisation
+  std::sort(map_list.begin(), map_list.end(), [](auto pair1, auto pair2) {
+    return pair1.first < pair2.first;
```

Comment:
Is it possible for two elements of the list to have the same first member? (In which case we'd also need to compare the second elements.)

Reply:
It's shouldn't be possible because the first elements are the keys in the result map.